### PR TITLE
New Serverless Pattern - Amazon EventBridge pipe - filter and transform using cdk python

### PR DESCRIPTION
**Pull request by manikandanks2000**
*Issue #, if available:*

*Description of changes:*
This pattern will use Amazon EventBridge Pipe to accept source messages from Amazon Simple Queue Service and will apply filter and transformation before sending the target message to AWS Step Functions.
This example project implemented with CDK/Python.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by manikandanks2000** [comment]
This branch is based on closed branch 
https://github.com/aws-samples/serverless-patterns/pull/1994
for some reason that branch was picking files outside my changes and was not able to remove them,  hence decided to create new one to avoid issues.  thanks for your support

File: cdk-eventbridge-pipes-sqs-to-stepfunctions-filter-transform-python/README.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+ aws sqs send-message \
+ --queue-url=SQS_URL \
+ --message-body '{"id":"id2","team": "Team2", "status": "COMPLETE"}'
+ 
+ aws sqs send-message \
+ --queue-url=SQS_URL \
+ --message-body '{"id":"id3","team": "Team3", "status": "NOTSTARTED"}'
+
+Validate the below from the step functions execution window.
```
[the remark this answers is not in the document]
Comment (bfreiberg) [reply cdk-eventbridge-pipes-sqs-to-stepfunctions-filter-transform-python/README.md:71]: Please add a link or description on how to reach that page

File: cdk-eventbridge-pipes-sqs-to-stepfunctions-filter-transform-python/cdk_eventbridge_pipes_sqs_to_stepfunctions_python/__init__.py
Comment (bfreiberg) [inline cdk-eventbridge-pipes-sqs-to-stepfunctions-filter-transform-python/cdk_eventbridge_pipes_sqs_to_stepfunctions_python/__init__.py:1]: Can the `__pycache__` be deleted?

File: cdk-eventbridge-pipes-sqs-to-stepfunctions-filter-transform-python/example-pattern.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
+{
+  "title": "EventBridge Pipe from SQS to Step Functions with filter and transformation",
+  "description": "Amazon EventBridge Pipe apply filter and transformation from Amazon Simple Queue Service to AWS Step Functions.",
+  "language": "Python",
+  "level": "300",
+  "framework": "CDK",
+  "introBox": {
+    "headline": "How it works",
+    "text": [
+      "The example will create an Amazon Simple Queue Service, AWS Step Functions and Amazon EventBridge Pipe.",
```
Comment (bfreiberg) [inline cdk-eventbridge-pipes-sqs-to-stepfunctions-filter-transform-python/example-pattern.json:10]:
```suggestion
      "The example will create an Amazon SQS queue, a AWS Step Functions workflow and an Amazon EventBridge Pipe.",
```

File: cdk-eventbridge-pipes-sqs-to-stepfunctions-filter-transform-python/example-pattern.json [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      "repoURL": "https://github.com/aws-samples/serverless-patterns/tree/main/cdk-eventbridge-pipes-sqs-to-stepfunctions-filter-transform-python",
+      "templateURL": "serverless-patterns/cdk-eventbridge-pipes-sqs-to-stepfunctions-filter-transform-python",
+      "projectFolder": "cdk_eventbridge_pipes_sqs_to_stepfunctions_python",
+      "templateFile": "cdk_eventbridge_pipes_sqs_to_stepfunctions_python_stack.py"
+    }
+  },
+  "resources": {
+    "bullets": [
+      {
+        "text": "Eventbridge Pipe filter and transformation",
```
Comment (bfreiberg) [inline cdk-eventbridge-pipes-sqs-to-stepfunctions-filter-transform-python/example-pattern.json:26]:
```suggestion
        "text": "Eventbridge Pipes filter and transformation",
```

File: cdk-eventbridge-pipes-sqs-to-stepfunctions-filter-transform-python/example-pattern.json [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    ]
+  },
+  "testing": {
+    "text": [
+      "See the GitHub repo for detailed testing instructions."
+    ]
+  },
+  "cleanup": {
+    "text": [
+      "Delete the stack: <code>cdk delete</code>."
```
Comment (bfreiberg) [inline cdk-eventbridge-pipes-sqs-to-stepfunctions-filter-transform-python/example-pattern.json:43]:
```suggestion
      "<code>cdk delete</code>"
```

**Comment by manikandanks2000** [comment]
@bfreiberg thanks for the inputs . I have accepted the recommendations and changes to the source.

**Comment by bfreiberg** [comment]
Thank you for the updates. Could you please remove the `__pycache__` folder.

**Comment by manikandanks2000** [comment]
> Thank you for the updates. Could you please remove the `__pycache__` folder.

Sure. I have deleted it now.

**Comment by bfreiberg** [comment]
Please use the correct folder name in your `example.json` and please use a shorter folder name for your CDK code as well, like `cdk` for example.

**Comment by manikandanks2000** [comment]
@bfreiberg project name modified as well.

**Comment by manikandanks2000** [comment]
I already have a variable aws_cdk as cdk in the python code, to avoid conflict have modified the folder name as "src" and stack file name as "cdkstack.py" . let me know if this is good or correcting it as cdk is your recommendation.

File: cdk-eventbridge-pipes-sqs-to-stepfunctions-filter-transform-python/example-pattern.json [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+      "Sending messages to the Amazon Simple Queue Service will trigger the pipe to start the AWS Step Function workflow execution.",
+      "SQS Messages will be filtered and transformed before sending to AWS Step Functions."
+    ]
+  },
+  "gitHub": {
+    "template": {
+      "repoURL": "https://github.com/aws-samples/serverless-patterns/tree/main/cdk-eventbridge-pipes-sqs-to-stepfunctions-filter-transform-python",
+      "templateURL": "serverless-patterns/cdk-eventbridge-pipes-sqs-to-stepfunctions-filter-transform-python",
+      "projectFolder": "eventbridge_pipes_sqs_to_stepfunctions_python",
+      "templateFile": "eventbridge_pipes_sqs_to_stepfunctions_python_stack.py"
```
Comment (bfreiberg) [inline cdk-eventbridge-pipes-sqs-to-stepfunctions-filter-transform-python/example-pattern.json:20]: Thanks for making the changes, please also adjust these references and make sure that the `templateFile` includes the folder name.

File: cdk-eventbridge-pipes-sqs-to-stepfunctions-filter-transform-python/example-pattern.json [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+      "The example will create an Amazon SQS queue, a AWS Step Functions workflow and an Amazon EventBridge Pipe.",
+      "Sending messages to the Amazon Simple Queue Service will trigger the pipe to start the AWS Step Function workflow execution.",
+      "SQS Messages will be filtered and transformed before sending to AWS Step Functions."
+    ]
+  },
+  "gitHub": {
+    "template": {
+      "repoURL": "https://github.com/aws-samples/serverless-patterns/tree/main/cdk-eventbridge-pipes-sqs-to-stepfunctions-filter-transform-python",
+      "templateURL": "serverless-patterns/cdk-eventbridge-pipes-sqs-to-stepfunctions-filter-transform-python",
+      "projectFolder": "src",
```
Comment (bfreiberg) [inline cdk-eventbridge-pipes-sqs-to-stepfunctions-filter-transform-python/example-pattern.json:19]:
```suggestion
      "projectFolder": "cdk-eventbridge-pipes-sqs-to-stepfunctions-filter-transform-python",
```

Comment (bfreiberg) [reply cdk-eventbridge-pipes-sqs-to-stepfunctions-filter-transform-python/example-pattern.json:19]: This is the folder name of the pattern, not the CDK code folder.

**Comment by manikandanks2000** [comment]
thanks for the guidance. hope it goes thru and will keep note of the changes recommended for my future contributions.

**Comment by bfreiberg** [comment]
Looks good, thank you for your contribution. A Developer Advocate will merge the PR soon.

**Pull request merged**